src/seraphis_mocks/enote_finding_context_mocks.h

```diff
@@ -102,6 +104,51 @@ class EnoteFindingContextLedgerMockLegacy final : public EnoteFindingContextLedg
     const LegacyScanMode m_legacy_scan_mode;
 };
 
+////
+// EnoteFindingContextMockLegacy
+// - find owned enotes from legacy view scanning using actual chain data
+///
+class EnoteFindingContextMockLegacy final
+{
+public:
+//constructors
+    EnoteFindingContextMockLegacy(
+        const rct::key &legacy_base_spend_pubkey,
+        const std::unordered_map<rct::key, cryptonote::subaddress_index> &legacy_subaddress_map,
+        const crypto::secret_key &legacy_view_privkey,
+        const std::function<bool(const cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::request&, cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::response&)> &_rpc_get_blocks):
+            m_legacy_base_spend_pubkey{legacy_base_spend_pubkey},
+            m_legacy_subaddress_map{legacy_subaddress_map},
+            m_legacy_view_privkey{legacy_view_privkey},
+            rpc_get_blocks{_rpc_get_blocks}
+    {
+    }
+
+//overloaded operators
+    /// disable copy/move (this is a scoped manager [reference wrapper])
+    EnoteFindingContextMockLegacy& operator=(EnoteFindingContextMockLegacy&&) = delete;
+
+//member functions
+    /// scans enotes in a tx for basic records
+    void find_basic_records(
+        const std::uint64_t block_index,
+        const std::uint64_t block_timestamp,
+        const rct::key &transaction_id,
+        const std::uint64_t total_enotes_before_tx,
+        const std::uint64_t unlock_time,
+        const TxExtra &tx_memo,
+        const std::vector<sp::LegacyEnoteVariant> &enotes,
+        std::list<sp::ContextualBasicRecordVariant> &collected_records) const;
+
+    /// abstracted function that gets blocks via RPC request
+    const std::function<bool(const cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::request&, cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::response&)> &rpc_get_blocks;
```

Comment:
Should this be marked `private`?

Reply:
I'm about to take a deeper pass at class structure and will probably make a lot of changes in line with this comment. Worth re-visiting after making those changes